Bio/SeqIO/AbiIO.py

```diff
@@ -406,11 +406,21 @@ def AbiIterator(handle, alphabet=None, trim=False):
         file_name = basename(handle.name).replace('.ab1', '')
     except AttributeError:
         file_name = ""
-    record = SeqRecord(Seq(seq, alphabet),
-                       id=sample_id, name=file_name,
-                       description='',
-                       annotations=annot,
-                       letter_annotations={'phred_quality': qual})
+    if handle.name.endswith('.fsa'):
```

Comment:
Very good point, came up with a different way of setting fsa details, set a try for record and if there is a NameError then try setting fsa